src/background.c

```diff
@@ -600,6 +600,44 @@ remove_outdated_instances (int stamp)
     }
 }
 
+static void
+update_background_monitor_properties (GHashTable *app_states)
+{
+  GVariantBuilder builder;
+  GHashTableIter iter;
+  InstanceData *data;
+  char *id;
+
+  g_variant_builder_init (&builder, G_VARIANT_TYPE_ARRAY);
```

Comment:
Perhaps `G_VARIANT_TYPE ("aa{sv}");` would be "safer"; e.g. I'm not sure what happens if the array is empty? Or will this some how be magically turned into the right type by the D-Bus marshaling code?

Reply:
Done